x/cash/dynamicfee.go

```diff
@@ -144,39 +109,61 @@ func (d DynamicFeeDecorator) chargeFee(store weave.KVStore, src weave.Address, a
 	return d.ctrl.MoveCoins(store, src, dest, amount)
 }
 
-// prepare is all shared setup between Check and Deliver, one more level above extractFee
-func (d DynamicFeeDecorator) prepare(ctx weave.Context, store weave.KVStore, tx weave.Tx) (fee x.Coin, payer weave.Address, cache weave.KVCacheWrap, err error) {
-	var finfo *FeeInfo
-	fee = x.Coin{}
+// chargeMinimalFee deduct an anty span fee from a given account.
+func (d DynamicFeeDecorator) chargeMinimalFee(store weave.KVStore, src weave.Address) error {
+	fee := gconf.Coin(store, GconfMinimalFee)
 
-	// extract expected fee
-	finfo, err = d.extractFee(ctx, tx, store)
+	if fee.IsZero() {
+		return nil
+	}
+
+	// If a fee has no currency set any ticker is accepted.
```

Comment:
q: now that I see this any ticker condition again. I am wondering if we really want this or something that allow us to define a list of options like gaiad: `Minimum gas prices to accept for transactions; Any fee in a tx must meet this minimum (e.g. 0.01photino;0.0001stake)`. the `;` is the OR but a `,` would be AND. I would see any solution in a separate PR of course but just want to highlight the additional complexity and uncertainty for a user which we have with this edge case.

Reply:
Added https://github.com/iov-one/weave/issues/369 for further discussions

Reply:
Adressed in https://github.com/iov-one/weave/pull/364/commits/5bdc3babe4b936726d84774564b7758710125673